examples/sparse/hgnn.py

```diff
@@ -0,0 +1,95 @@
+"""
+Hypergraph Neural Networks (https://arxiv.org/pdf/1809.09401.pdf)
+"""
+import dgl
+import dgl.data
+import dgl.mock_sparse
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchmetrics.functional import accuracy
+import tqdm
+
+dataset = dgl.data.CoraGraphDataset()
+
+graph = dataset[0]
+# The paper created the hypergraph by "each time one vertex in the
+# graph is selected as the centroid and its connected vertices
+# are used to generate one hyperedge including the centroid
+# itself".  In this case, the incidence matrix of the hypergraph
+# is the same as the adjacency matrix of the original graph (with
+# self-loops).
+# We follow the paper assuming that the rows of the incidence matrix
+# are nodes and the columns are edges.
+
+# QUESTION: can we either (1) easily convert a PyTorch sparse matrix
+# to a DGL sparse matrix, or (2) make g.adj() a DGL sparse matrix
+# in the future?
```

Comment:
L25-27 has been tracked by https://github.com/dmlc/dgl/issues/4876. You may remove these comments.